### PR TITLE
BUGFIX Stream Registry - Cannot Query Streams by their schema

### PR DESCRIPTION
**Pull request by rshotels**
Fixed issue in StreamFilter

# stream-registry PR

BUGFIX Stream Registry - Cannot Query Streams by their schema
moved common filtering into FilterUtility
fixed issue in StreamFilter
# PR Checklist Forms

- [ ] CHANGELOG.md updated
- [ ] Reviewer assigned
- [ ] PR assigned (presumably to submitter)
- [ ] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: it/src/test/java/com/expediagroup/streamplatform/streamregistry/it/StreamTestStage.java [hunk 1]
```diff
@@ -115,15 +115,15 @@ public void queryByRegex() {
 
     setFactorySuffix("query_by_regex");
 
-    StreamKeyQuery query = StreamKeyQuery.builder().nameRegex("stream_name.*").build();
+    StreamKeyQuery query = StreamKeyQuery.builder().build();//.nameRegex(".*query_by_regex.*").build();
```
Comment (nahguam) [inline it/src/test/java/com/expediagroup/streamplatform/streamregistry/it/StreamTestStage.java]: remove comment

File: graphql-api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/StreamFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    stream.setSpecification(matchingSpecification());
+    stream.getSpecification().getTags().get(0).setName(FAIL);
+    assertFalse(streamFilter.test(stream));
+
+    stream.setSpecification(matchingSpecification());
+    stream.getSpecification().getTags().get(0).setValue(FAIL);
+    assertFalse(streamFilter.test(stream));
+
+    //nulls
```
Comment (nahguam) [inline graphql-api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/StreamFilterTest.java]: ?

File: graphql-api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/SchemaFilterTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.expediagroup.streamplatform.streamregistry.graphql.model.queries.SchemaKeyQuery;
+import com.expediagroup.streamplatform.streamregistry.graphql.model.queries.SpecificationQuery;
+import com.expediagroup.streamplatform.streamregistry.model.Schema;
+import com.expediagroup.streamplatform.streamregistry.model.keys.SchemaKey;
+
+public class SchemaFilterTest {
+
+  @Test
+  public void test1() {
```
Comment (nahguam) [inline graphql-api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/SchemaFilterTest.java]: Please use a more description test name

File: graphql-api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/StreamFilterTest.java (hunk 2 above)
Comment (rshotels) [reply graphql-api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/StreamFilterTest.java]: Hoist by my own petard. 
I shall remove all comments and change function names as requested.

File: graphql-api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/SchemaFilterTest.java (hunk 3 above)
Comment (rshotels) [reply graphql-api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/SchemaFilterTest.java]: I shall

Comment (rshotels) [reply graphql-api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/filters/SchemaFilterTest.java]: there's also no assertion...

**Pull request merged**